### ISSUE TITLE
maia-wasm waterfall panics when the horizontal resolution is large

### ISSUE THREAD

**Issue by DEFAULTTEXHAPb** [maia-sdr/maia-sdr#79]
After installing maia-sdr firmware from release as [instruction](https://maia-sdr.org/installation/) describes i cannot record spectrogram. Running `./maia-httpd` from `/root` directory returns this error
```console
# ./maia-httpd
Error: failed to open maia-sdr-spectrometer DMA buffer

Caused by:
    0: failed to open rxbuffer DMA buffer
    1: mmap rxbuffer failed
```
I did not change my default IP address as described in "Configure the Pluto USB Ethernet". So i opened connection to Pluto on 192.168.3.1 IP-address from Mozilla Firefox on Windows 10. Opening `http://192.168.3.1:8000/` shows spectrogram screen with black screen as this

![image](https://github.com/maia-sdr/plutosdr-fw/assets/71003801/cc721f90-e3d1-4e2f-94b7-b22764837103)

I've flashed pluto using `pluto.frm` file from release archive. I am using ADALM-PLUTO REV B. `fw_printenv` command outputs this

```console
# fw_printenv
adi_loadvals=fdt addr ${fit_load_address} && fdt get value fdt_choosen /configurations/${fit_config}/ fdt && fdt get addr fdtaddr /images/${fdt_choosen} data && fdt addr ${fdtaddr}; if test -n ${ad936x_ext_refclk}  && test ! -n ${ad936x_skip_ext_refclk}; then fdt set /clocks/clock@0 clock-frequency ${ad936x_ext_refclk}; fi; if test -n ${model}; then fdt set / model ${model}; fi; if test -n ${attr_name} && test -n ${attr_val}; then fdt set /amba/spi@e0006000/ad9361-phy@0 ${attr_name} ${attr_val}; fi
baudrate=115200
bitstream_image=system.bit.bin
bitstream_size=0x400000
boot_image=BOOT.bin
boot_size=0xF00000
bootcmd=run $modeboot
bootdelay=0
bootenv=uEnv.txt
clear_reset_cause=mw f8000008 df0d && mw f8000258 00400000 && mw f8000004 767b
devicetree_image=devicetree.dtb
devicetree_load_address=0x2000000
devicetree_size=0x20000
dfu_alt_info=boot.dfu raw 0x0 0x100000;firmware.dfu raw 0x200000 0x1E00000;uboot-extra-env.dfu raw 0xFF000 0x1000;uboot-env.dfu raw 0x100000 0x20000;spare.dfu raw 0x120000 0xE0000
dfu_ram=echo Entering DFU RAM mode ... && run dfu_ram_info && dfu 0 ram 0
dfu_ram_info=set dfu_alt_info dummy.dfu ram 0 0\\;firmware.dfu ram ${fit_load_address} 0x1E00000
dfu_sf=gpio set 15;set stdout serial@e0001000;echo Entering DFU SF mode ... && run dfu_sf_info && dfu 0 sf 0:0:40000000:0 && if test -n ${dfu_alt_num} && test ${dfu_alt_num} = 1; then set fit_size ${filesize} && set dfu_alt_num && env save; fi;gpio clear 15;
dfu_sf_info=set dfu_alt_info boot.dfu raw 0x0 0x100000\\;firmware.dfu raw 0x200000 0x1E00000\\;uboot-extra-env.dfu raw 0xFF000 0x1000\\;uboot-env.dfu raw 0x100000 0x20000\\;spare.dfu raw 0x120000 0xE0000
ethaddr=00:0a:35:00:01:22
extraenv_load_address=0x207E000
fdt_high=0x20000000
filesize=9a620b
fit_config=config@0
fit_load_address=0x2080000
importbootenv=echo Importing environment from SD ...; env import -t ${loadbootenv_addr} $filesize
initrd_high=0x20000000
jtagboot=env default -a;sf probe && sf protect unlock 0 100000 && run dfu_sf;
kernel_image=uImage
loadbit_addr=0x100000
loadbootenv=load mmc 0 ${loadbootenv_addr} ${bootenv}
loadbootenv_addr=0x2000000
maxcpus=1
preboot=if test $modeboot = sdboot && env run sd_uEnvtxt_existence_test; then if env run loadbootenv; then env run importbootenv; fi; fi;
qspiboot_extraenv=sf read ${extraenv_load_address} 0xFF000 0x1000 && env import -c ${extraenv_load_address} 0x1000 || true
ramdisk_image=uramdisk.image.gz
ramdisk_load_address=0x4000000
ramdisk_size=0x400000
read_sf=sf probe 0:0 50000000 0 && run qspiboot_extraenv &&sf read ${fit_load_address} 0x200000 ${fit_size} && iminfo ${fit_load_address} || sf read ${fit_load_address} 0x200000  0x1E00000;
sd_uEnvtxt_existence_test=test -e mmc 0 /uEnv.txt
stdout=serial@e0001000
thor_ram=run dfu_ram_info && thordown 0 ram 0
uenvboot=if run loadbootenv; then echo Loaded environment from ${bootenv}; run importbootenv; fi; if test -n $uenvcmd; then echo Running uenvcmd ...; run uenvcmd; fi
usbboot=if usb start; then run uenvboot; echo Copying Linux from USB to RAM... && load usb 0 ${fit_load_address} ${kernel_image} && load usb 0 ${devicetree_load_address} ${devicetree_image} && load usb 0 ${ramdisk_load_address} ${ramdisk_image} && bootm ${fit_load_address} ${ramdisk_load_address} ${devicetree_load_address}; fi
hostname=pluto
ipaddr=192.168.3.1
ipaddr_host=192.168.3.10
netmask=255.255.255.0
udc_handle_suspend=0
netmask_eth=255.255.255.0
ramboot_verbose=adi_hwref;echo Copying Linux from DFU to RAM... && run dfu_ram;if run adi_loadvals; then echo Loaded AD936x refclk frequency and model into devicetree; fi; envversion;setenv bootargs console=ttyPS0,115200 maxcpus=${maxcpus} rootfstype=ramfs root=/dev/ram0 rw earlyprintk clk_ignore_unused uio_pdrv_genirq.of_id=uio_pdrv_genirq uboot="${uboot-version}" && bootm ${fit_load_address}#${fit_config}
qspiboot_verbose=adi_hwref;echo Copying Linux from QSPI flash to RAM... && run read_sf && if run adi_loadvals; then echo Loaded AD936x refclk frequency and model into devicetree; fi; envversion;setenv bootargs console=ttyPS0,115200 maxcpus=${maxcpus} rootfstype=ramfs root=/dev/ram0 rw earlyprintk clk_ignore_unused uio_pdrv_genirq.of_id=uio_pdrv_genirq uboot="${uboot-version}" && bootm ${fit_load_address}#${fit_config} || echo BOOT failed entering DFU mode ... && run dfu_sf
qspiboot=set stdout nulldev;adi_hwref;test -n $PlutoRevA || gpio input 14 && set stdout serial@e0001000 && sf probe && sf protect lock 0 100000 && run dfu_sf;  set stdout serial@e0001000;itest *f8000258 == 480003 && run clear_reset_cause && run dfu_sf; itest *f8000258 == 480007 && run clear_reset_cause && run ramboot_verbose; itest *f8000258 == 480006 && run clear_reset_cause && run qspiboot_verbose; itest *f8000258 == 480002 && run clear_reset_cause && exit; echo Booting silently && set stdout nulldev; run read_sf && run adi_loadvals; envversion;setenv bootargs console=ttyPS0,115200 maxcpus=${maxcpus} rootfstype=ramfs root=/dev/ram0 rw quiet loglevel=4 clk_ignore_unused uio_pdrv_genirq.of_id=uio_pdrv_genirq uboot="${uboot-version}" && bootm ${fit_load_address}#${fit_config} || set stdout serial@e0001000;echo BOOT failed entering DFU mode ... && sf protect lock 0 100000 && run dfu_sf
attr_name=compatible
attr_val=ad9364
mode=1r1t
fit_size=C695A7
```

How to solve this? I am an inexperienced user of Pluto.

**Comment by daniestevez** [comment]
I have opened a pull request with a change that should fix this problem for resolutions smaller than ~5500px. You can test the firmware built from the pull request by the CI: https://github.com/maia-sdr/maia-sdr/actions/runs/6817809930

**Comment by DEFAULTTEXHAPb** [comment]
I tested this firmware. When waterfall is opened on full resolution, console throws this error

```
panicked at 'assertion failed: 2 * freqs.len() <= (1 << 16)', src/waterfall.rs:503:9

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://192.168.3.1:8000/pkg/maia_wasm.js:372:21
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[680]:0x418d1
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[365]:0x2ee41
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[504]:0x3f34f
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[485]:0x33762
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[176]:0x7f42
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[502]:0x360bf
maia_wasm_start@http://192.168.3.1:8000/pkg/maia_wasm.js:280:14
run@http://192.168.3.1:8000/index.js:5:5


[maia_wasm.js:388:21](http://192.168.3.1:8000/pkg/maia_wasm.js)
    __wbg_error_f851667af71bcfc6 http://192.168.3.1:8000/pkg/maia_wasm.js:388
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:268696
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:192065
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:258895
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:210786
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:32578
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:221375
    maia_wasm_start http://192.168.3.1:8000/pkg/maia_wasm.js:280
    run http://192.168.3.1:8000/index.js:5

Uncaught (in promise) RuntimeError: unreachable executed
    maia_wasm_start http://192.168.3.1:8000/pkg/maia_wasm.js:280
    run http://192.168.3.1:8000/index.js:5

```

Waterfall starts fine on small window size, but after resizing window to full-screen console throws an error

```
panicked at 'already borrowed: BorrowMutError', src/ui.rs:316:24

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://192.168.3.1:8000/pkg/maia_wasm.js:372:21
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[680]:0x418d1
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[365]:0x2ee41
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[504]:0x3f37d
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[416]:0x31339
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[178]:0xb81c
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[268]:0x25d24
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[545]:0x3fd17
__wbg_adapter_44@http://192.168.3.1:8000/pkg/maia_wasm.js:255:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:236:20
promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb@http://192.168.3.1:8000/pkg/maia_wasm.js:1013:37
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[610]:0x40a2a
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[312]:0x2ab56
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[362]:0x2eb40
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[308]:0x2a685
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[466]:0x331d6
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[545]:0x3fd17
__wbg_adapter_44@http://192.168.3.1:8000/pkg/maia_wasm.js:255:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:236:20
promise callback*__wbg_get_imports/imports.wbg.__wbg_then_b2267541e2a73865@http://192.168.3.1:8000/pkg/maia_wasm.js:1017:37
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[225]:0x1d68c
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[316]:0x2afd4
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[178]:0x93a2
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[268]:0x25d24
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[545]:0x3fd17
__wbg_adapter_44@http://192.168.3.1:8000/pkg/maia_wasm.js:255:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:236:20
promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb@http://192.168.3.1:8000/pkg/maia_wasm.js:1013:37
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[610]:0x40a2a
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[312]:0x2ab56
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[362]:0x2eb40
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[308]:0x2a685
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[466]:0x331d6
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[545]:0x3fd17
__wbg_adapter_44@http://192.168.3.1:8000/pkg/maia_wasm.js:255:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:236:20
promise callback*__wbg_get_imports/imports.wbg.__wbg_then_b2267541e2a73865@http://192.168.3.1:8000/pkg/maia_wasm.js:1017:37
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[225]:0x1d68c
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[178]:0x92fa
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[268]:0x25d24
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[545]:0x3fd17
__wbg_adapter_44@http://192.168.3.1:8000/pkg/maia_wasm.js:255:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:236:20
promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb@http://192.168.3.1:8000/pkg/maia_wasm.js:1013:37
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[610]:0x40a2a
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[455]:0x32d49
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[381]:0x2fac2
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[538]:0x3fc34
__wbg_adapter_323@http://192.168.3.1:8000/pkg/maia_wasm.js:299:10
cb0@http://192.168.3.1:8000/pkg/maia_wasm.js:997:28
__wbg_get_imports/imports.wbg.__wbg_new_43f1b47c28813cbd@http://192.168.3.1:8000/pkg/maia_wasm.js:1002:25
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[674]:0x4165f
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[420]:0x314e1
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[550]:0x3fdb1
__wbg_adapter_27@http://192.168.3.1:8000/pkg/maia_wasm.js:218:22
real@http://192.168.3.1:8000/pkg/maia_wasm.js:200:20


[maia_wasm.js:388:21](http://192.168.3.1:8000/pkg/maia_wasm.js)
    __wbg_error_f851667af71bcfc6 http://192.168.3.1:8000/pkg/maia_wasm.js:388
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:268696
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:192065
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:258941
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:201529
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:47132
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:154916
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261399
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    (Async: promise callback)
    __wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:264746
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:174934
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:191296
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:173701
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:209366
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261399
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    (Async: promise callback)
    __wbg_then_b2267541e2a73865 http://192.168.3.1:8000/pkg/maia_wasm.js:1017
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:120460
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:176084
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:37794
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:154916
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261399
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    (Async: promise callback)
    __wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:264746
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:174934
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:191296
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:173701
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:209366
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261399
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    (Async: promise callback)
    __wbg_then_b2267541e2a73865 http://192.168.3.1:8000/pkg/maia_wasm.js:1017
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:120460
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:37626
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:154916
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261399
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    (Async: promise callback)
    __wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:264746
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:208201
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:195266
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261172
    __wbg_adapter_323 http://192.168.3.1:8000/pkg/maia_wasm.js:299
    cb0 http://192.168.3.1:8000/pkg/maia_wasm.js:997
    __wbg_new_43f1b47c28813cbd http://192.168.3.1:8000/pkg/maia_wasm.js:1002
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:267871
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:201953
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261553
    __wbg_adapter_27 http://192.168.3.1:8000/pkg/maia_wasm.js:218
    real http://192.168.3.1:8000/pkg/maia_wasm.js:200

Uncaught (in promise) RuntimeError: unreachable executed
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    promise callback*__wbg_get_imports/imports.wbg.__wbg_then_b2267541e2a73865 http://192.168.3.1:8000/pkg/maia_wasm.js:1017
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    promise callback*__wbg_get_imports/imports.wbg.__wbg_then_b2267541e2a73865 http://192.168.3.1:8000/pkg/maia_wasm.js:1017
    __wbg_adapter_44 http://192.168.3.1:8000/pkg/maia_wasm.js:255
    real http://192.168.3.1:8000/pkg/maia_wasm.js:236
    promise callback*__wbg_get_imports/imports.wbg.__wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013
    __wbg_adapter_323 http://192.168.3.1:8000/pkg/maia_wasm.js:299
    cb0 http://192.168.3.1:8000/pkg/maia_wasm.js:997
    __wbg_new_43f1b47c28813cbd http://192.168.3.1:8000/pkg/maia_wasm.js:1002
    __wbg_adapter_27 http://192.168.3.1:8000/pkg/maia_wasm.js:218
    real http://192.168.3.1:8000/pkg/maia_wasm.js:200
[maia_wasm_bg.wasm:192103:1](http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm)
    real http://192.168.3.1:8000/pkg/maia_wasm.js:242
    (Async: promise callback)
    __wbg_then_f7e06ee3c11698eb http://192.168.3.1:8000/pkg/maia_wasm.js:1013

panicked at 'already borrowed: BorrowMutError', src/websocket.rs:64:14

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://192.168.3.1:8000/pkg/maia_wasm.js:372:21
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[680]:0x418d1
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[365]:0x2ee41
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[504]:0x3f37d
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[416]:0x31339
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[256]:0x244fd
@http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:wasm-function[540]:0x3fc79
__wbg_adapter_24@http://192.168.3.1:8000/pkg/maia_wasm.js:214:10
real@http://192.168.3.1:8000/pkg/maia_wasm.js:200:20


[maia_wasm.js:388:21](http://192.168.3.1:8000/pkg/maia_wasm.js)
    __wbg_error_f851667af71bcfc6 http://192.168.3.1:8000/pkg/maia_wasm.js:388
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:268696
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:192065
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:258941
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:201529
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:148733
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261241
    __wbg_adapter_24 http://192.168.3.1:8000/pkg/maia_wasm.js:214
    real http://192.168.3.1:8000/pkg/maia_wasm.js:200
Uncaught RuntimeError: unreachable executed
    __wbg_adapter_24 http://192.168.3.1:8000/pkg/maia_wasm.js:214
    real http://192.168.3.1:8000/pkg/maia_wasm.js:200
[maia_wasm_bg.wasm:192103:1](http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm)
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:192103
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:258941
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:201529
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:148733
    <anonymous> http://192.168.3.1:8000/pkg/maia_wasm_bg.wasm:261241
    __wbg_adapter_24 http://192.168.3.1:8000/pkg/maia_wasm.js:214
    real http://192.168.3.1:8000/pkg/maia_wasm.js:200
```

**Comment by daniestevez** [comment]
What is the screen resolution and sample rate that you're using?

Initially I only did tests at 61.44 Msps on the PR, and found that it worked up to a resolution of ~5500px. Now I'm testing other sample rates. For instance, 40 Msps only works up to 3696px.

**Comment by DEFAULTTEXHAPb** [comment]
Sorry for not answering so long. The screen resolution is 2560 x 1440 with  125% scaling, 60 Hz refresh rate. I tested app on 61.44 Msps and 40 Msps, but i can not repeat the error. Waterfall works fine.

**Comment by daniestevez** [comment]
In any case, I have pushed a new commit to the pull request that limits the number of frequency labels instead of crashing if the screen resolution is very large.

**Comment by daniestevez** [comment]
I'm going to close this, since the fix is already in v0.6.1. If a similar problem appears, feel free to re-open the issue.
